src/nupic/regions/knn_anomaly_classifier_region.py

```diff
@@ -827,6 +835,125 @@ def removeLabels(self, start=None, end=None, labelFilter=None):
+
+      if "relativeThreshold" in knnParams:
+        knnParams["relativeThreshold"] = bool(knnParams["relativeThreshold"])
+
+      if "doSphering" in knnParams:
+        knnParams["doSphering"] = bool(knnParams["doSphering"])
+
+      if "replaceDuplicates" in knnParams:
+        knnParams["replaceDuplicates"] = bool(knnParams["replaceDuplicates"])
+
+      proto.knnclassifierArgs = knnParams
+
+    if self._knnclassifier is not None:
+      self._knnclassifier.writeToProto(proto.knnclassifier)
+    if self.labelResults is not None:
+      proto.labelResults = self.labelResults
+    if self.saved_categories is not None:
+      proto.savedCategories = self.saved_categories
+    if self._recordsCache is not None:
+      recordsCache = proto.init("recordsCache", len(self._recordsCache))
+      for i, item in enumerate(self._recordsCache):
+        record = recordsCache[i]
+        record.rowid = int(item.ROWID)
+        record.anomalyScore = float(item.anomalyScore)
+        if item.anomalyVector is not None:
+          record.anomalyVector = item.anomalyVector
+        if item.anomalyLabel is not None:
+          record.anomalyLabel = item.anomalyLabel
+        record.setByUser = bool(item.setByUser)
+
+
+  @classmethod
+  def readFromProto(cls, proto):
+    if proto.version != KNNAnomalyClassifierRegion.__VERSION__:
+      raise RuntimeError("Invalid KNNAnomalyClassifierRegion Version")
+
+    instance = object.__new__(cls)
+    instance._version = proto.version
+    instance._maxLabelOutputs = proto.maxLabelOutputs
+    instance._activeColumnCount = proto.activeColumnCount
+    instance._prevPredictedColumns = numpy.array(proto.prevPredictedColumns)
+    instance._anomalyVectorLength = proto.anomalyVectorLength
+    instance._classificationMaxDist = proto.classificationMaxDist
+    instance._iteration = proto.iteration
+    instance.trainRecords = proto.trainRecords
+    instance.anomalyThreshold = proto.anomalyThreshold
+    instance.cacheSize = proto.cacheSize
+    instance.classificationVectorType = proto.classificationVectorType
+    instance._knnclassifierArgs = proto.knnclassifierArgs.to_dict()
+
+    # Convert keys
+    if "svdSampleCount" in instance._knnclassifierArgs:
+      SVDSampleCount = instance._knnclassifierArgs["svdSampleCount"]
+      del instance._knnclassifierArgs["svdSampleCount"]
+      instance._knnclassifierArgs["SVDSampleCount"] = SVDSampleCount
+    if "svdDimCount" in instance._knnclassifierArgs:
+      SVDDimCount = instance._knnclassifierArgs["svdDimCount"]
+      del instance._knnclassifierArgs["svdDimCount"]
+      instance._knnclassifierArgs["SVDDimCount"] = SVDDimCount
+
+    instance._knnclassifier = KNNClassifierRegion.readFromProto(
+      proto.knnclassifier)
+    instance.labelResults = list(proto.labelResults)
+    instance.saved_categories = list(proto.savedCategories)
+
+    instance._recordsCache = []
+    for i, item in enumerate(proto.recordsCache):
```

Comment:
Doesn't look like you are using `i` so you can drop `enumerate` and just iterate over the cache items